### PR TITLE
Issues/383

### PR DESCRIPTION
**Pull request by smcgrath0**
Thanks for contributing to Peregrine CMS!

**What does this implement/fix? Explain your changes.**

Tags component should show display value instead of the tag name

**Does this close any currently open issues?**

#383 

**Any other comments?**

**Where has this been tested?**
Local 
Screen Shot 2020-05-14 at 4 02 25 PM

*Brower (version):*
Chrome Mac

### CODE REVIEW

File: pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java [hunk 1]
```diff
@@ -320,7 +320,13 @@ public Tag(Resource r) {
             this.path = r.getPath();
             this.path = path.substring(path.indexOf("/jcr:content"));
             this.name = r.getName();
-            this.value = r.getValueMap().get("value", String.class);
+            String tag = r.getValueMap().get("value", String.class);
+            Resource tagResource = r.getResourceResolver().getResource(tag);
+            this.value = tagResource.getValueMap().get("value", String.class);
+
+            if (this.value == null) {
+                this.value = "Please Configure";
```
Comment (reusr1) [inline pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java]: I dont think please configure is a good value for no value - that part should be left to the UI to render (a fallback to the name when rendering the tag would be better)

File: pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java [hunk 2]
```diff
@@ -320,7 +320,13 @@ public Tag(Resource r) {
             this.path = r.getPath();
             this.path = path.substring(path.indexOf("/jcr:content"));
             this.name = r.getName();
-            this.value = r.getValueMap().get("value", String.class);
+            String tag = r.getValueMap().get("value", String.class);
+            Resource tagResource = r.getResourceResolver().getResource(tag);
+            this.value = tagResource.getValueMap().get("value", String.class);
+
```
Comment (reusr1) [inline pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java:336]: if you delete a tag after you added the tag to the page this code will fail with a null pointer exception.

File: pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java [hunk 3]
```diff
@@ -320,7 +320,13 @@ public Tag(Resource r) {
             this.path = r.getPath();
             this.path = path.substring(path.indexOf("/jcr:content"));
             this.name = r.getName();
```
Comment (reusr1) [inline pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java:330]: looks like this is a former error. the name of the tag should be fetched from the tag itself (same place as the value) and not from the name of the node that references it (probably fails to render the tag name correctly if the name of the tag ever changes)

**Review by reusr1 — CHANGES_REQUESTED** [review]

File: pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java (hunk 3 above)
Comment (smcgrath0) [reply pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java:330]: That is correct, got objects working when deleted, can't use/see them. One other problem, cannot save/edit tags once placed on page. Think it has to do with our tags.infinity.json and how the data is stored, the tag objects are still there after deleting. 
Screen Shot 2020-05-15 at 1 23 53 PM

File: pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java [hunk 4]
```diff
@@ -254,7 +254,12 @@ public String getDataDefault() {
         List<Tag> answer = new ArrayList<Tag>();
         if(tags != null) {
             for(Resource tag: tags.getChildren()) {
-                answer.add(new Tag(tag));
+                String tagString = tag.getValueMap().get("value", String.class);
+                Resource tagResource = tag.getResourceResolver().getResource(tagString);
+                if (tagResource != null) { 
+                // if (tag != null) {
```
Comment (reusr1) [inline pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java]: no need to keep commented out code here

Comment (smcgrath0) [reply pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java]: Just pushed up without the comment

**Comment by smcgrath0** [comment]
Should be good to go @reusr1

File: pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java [hunk 5]
```diff
@@ -320,7 +328,15 @@ public Tag(Resource r) {
...
-            this.value = r.getValueMap().get("value", String.class);
+            String tag = r.getValueMap().get("value", String.class);
+            Resource tagResource = r.getResourceResolver().getResource(tag);
+            if (tagResource != null) {
+                this.value = tagResource.getValueMap().get("value", String.class);
+            }
+
+            if (this.value == null) {
+                this.value = null;
+            }
```
Comment (reusr1) [inline pagerenderer/vue/core/src/main/java/com/peregrine/pagerender/vue/models/PageModel.java:339]: not sure if this code really makes sense

**Pull request closed without merging**